src/registry/mod.rs

```diff
@@ -108,4 +109,41 @@ impl RegistryUri {
 
         Ok(())
     }
+
+    /// Returns the type of registry this URI refers to
+    pub fn kind(&self) -> RegistryType {
```

Comment:
imo this can just go completely

Reply:
It can't, as we need a way to map URIs to the correct Registry implementations when instancing them. This is mostly relevant for RegistryProvider, and is one of the major benefits of implementing that struct, as it hides away the complexity of having multiple registry types.